rules/windows/process_creation/proc_creation_win_code_devtunnel_tunneling.yaml

```
@@ -0,0 +1,37 @@
+title: Visual Studio Code/DevTunnel Abuse Detection
+id: 12345678-1234-5678-9123-456789abcdef
+description: |
+  Detects suspicious usage of Visual Studio Code's tunneling features, often abused in espionage campaigns.
+References:
+    - https://unit42.paloaltonetworks.com/stately-taurus-abuses-vscode-southeast-asian-espionage/
+status: experimental
+author: 0xAnalyst
+date: 2024/08/14
+logsource:
+  product: windows
+  service: process_creation
+detection:
+  selection:
+    - ProcessVersionInfoProductName: "Visual Studio Code"
+      CommandLine|contains: "tunnel"
+    - CommandLine|all:
+        - "host"
+        - "allow-anonymous"
+    - CommandLine|all:
+        - "port"
+        - "create"
+        - "-p"
+  condition: selection
```

```suggestion
detection:
    selection_cli:
        ProcessVersionInfoProductName: "Visual Studio Code"
        CommandLine|contains: "tunnel"
     selection_temporary:
         CommandLine|all:
             - "host"
             - "allow-anonymous"
    selection_persistent:
        CommandLine|all:
            - "port"
            - "create"
            - "-p"
  condition: 1 of selection_*
```
I did not change the logic but make it easier to grasp